src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    public static final int SALARY_INDEX = 2;
+    public static final int SALARY_MULTIPLIER_INDEX = 3;
+    public static final int DATE_INDEX = 0;
+    public static final String DATA_SEPARATOR = " ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder appender = new StringBuilder("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        LocalDate dateFromParsed = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate dateToParsed = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+
+        for (String name : names) {
+            int sum = 0;
+
+            for (String row : data) {
+                String[] info = row.split(DATA_SEPARATOR);
+
+                if (!info[1].equals(name)) {
+                    continue;
+                }
```

Comment:
why do you need this?

Reply:
To not make addition to salary sum if name is not for required person